readthedocs/core/utils/tasks/permission_checks.py

```diff
@@ -0,0 +1,9 @@
+__all__ = ('user_id_matches',)
+
+
+def user_id_matches(request, state, context):
```

Comment:
We have some logic around this kind of stuff in the privacy app as well -- so it might make sense to put it there.